images/virtualization-artifact/pkg/controller/cvi/internal/inuse_test.go

```diff
@@ -0,0 +1,383 @@
+					Kind: virtv2.ClusterVirtualImageKind,
+					Name: "test",
+				},
+			}),
+			generateVDForInUseTest("test2", "ns2", virtv2.VirtualDiskDataSource{
+				Type: virtv2.DataSourceTypeObjectRef,
+				ObjectRef: &virtv2.VirtualDiskObjectRef{
+					Kind: virtv2.ClusterVirtualImageKind,
+					Name: "test",
+				},
+			}),
+			generateVDForInUseTest("test3", "ns3", virtv2.VirtualDiskDataSource{
+				Type: virtv2.DataSourceTypeObjectRef,
+				ObjectRef: &virtv2.VirtualDiskObjectRef{
+					Kind: virtv2.ClusterVirtualImageKind,
+					Name: "test",
+				},
+			}),
+			generateVDForInUseTest("test4", "ns4", virtv2.VirtualDiskDataSource{
+				Type: virtv2.DataSourceTypeObjectRef,
+				ObjectRef: &virtv2.VirtualDiskObjectRef{
+					Kind: virtv2.ClusterVirtualImageKind,
+					Name: "test",
+				},
+			}),
+		},
+		VIs: []virtv2.VirtualImage{
+			generateVIForInUseTest("test", "ns", virtv2.VirtualImageDataSource{
+				Type: virtv2.DataSourceTypeObjectRef,
+				ObjectRef: &virtv2.VirtualImageObjectRef{
+					Kind: virtv2.ClusterVirtualImageKind,
+					Name: "test",
+				},
+			}),
+			generateVIForInUseTest("test2", "ns5", virtv2.VirtualImageDataSource{
+				Type: virtv2.DataSourceTypeObjectRef,
+				ObjectRef: &virtv2.VirtualImageObjectRef{
+					Kind: virtv2.ClusterVirtualImageKind,
+					Name: "test",
+				},
+			}),
+		},
+		CVIs: []virtv2.ClusterVirtualImage{
+			generateCVIForInUseTest("test2", virtv2.ClusterVirtualImageDataSource{
+				Type: virtv2.DataSourceTypeObjectRef,
+				ObjectRef: &virtv2.ClusterVirtualImageObjectRef{
+					Kind: virtv2.ClusterVirtualImageKind,
+					Name: "test",
+				},
+			}),
+			*cvibuilder.New(
+				cvibuilder.WithName("test322"),
+				cvibuilder.WithPhase(virtv2.ImageReady),
+				cvibuilder.WithCondition(metav1.Condition{
+					Status: metav1.ConditionTrue,
+					Reason: cvicondition.ReadyType.String(),
+				}),
+				cvibuilder.WithDatasource(virtv2.ClusterVirtualImageDataSource{
+					Type: virtv2.DataSourceTypeObjectRef,
+					ObjectRef: &virtv2.ClusterVirtualImageObjectRef{
+						Kind: virtv2.ClusterVirtualImageKind,
+						Name: "test",
+					},
+				}),
+			),
+		},
+		VMBDAs: []virtv2.VirtualMachineBlockDeviceAttachment{
+			generateVMBDAForInUseTest(
+				"test",
+				"ns",
+				virtv2.VMBDAObjectRef{
+					Kind: virtv2.ClusterVirtualImageKind,
+					Name: "test",
+				},
+				"vm",
+			),
+		},
+		ExpectedConditionExists:  true,
+		ExpectedConditionStatus:  metav1.ConditionTrue,
+		ExpectedConditionReason:  cvicondition.InUse.String(),
+		ExpectedConditionMessage: "5 VirtualMachines are using the ClusterVirtualImage, the ClusterVirtualImage is currently being used by the VMBDA ns/test, the ClusterVirtualImage is currently used to create 4 VirtualDisks, the ClusterVirtualImage is currently being used to create the VirtualImages: ns/test, ns5/test2, the ClusterVirtualImage is currently being used to create the ClusterVirtualImage test2, the ClusterVirtualImage is currently using in 6 Namespaces.",
```

Comment:
This one is noisy and unreadable :(

Also, "in use in 6 Namespaces" is not informative. If we want to reduce information and display only a count for more than 3 objects, can we display first 3 objects?

```
the ClusterVirtualImage is used by objects in Namespace/vm, Namespace/vm-databases, Namespace/vm-php and 3 more Namespaces."
```

Also, why duplicate "the ClusterVirtualImage" and following verbs? It can be used once:

```
The ClusterVirtualImage is still in use. VirtualMachines: ns/vm01, ns2/worker-01, ns3/worker-02, and 2 more. VirtualMachineBlockDeviceAttachment: ns/test. VirtualDisks ns/vm-01-root, ns2/worker-01-data, ns2/worker-01-root, and 1 more. VirtualImages: ns/test, ns5/test2. ClusterVirtualImage test2. Namespaces: ns, ns1, ns2, and 3 more.
```